### ISSUE TITLE
[MVP] Focus-Handling - Special case selecting first bundle item in case of mandatory  attribute

### ISSUE THREAD

**Issue by Uli-Tiger** [SAP/spartacus#11657]
In case a user adds/selects the first bundle item of an mandatory attribute, the add/select button will disappear and no remove/deselect button appears. hence we can't restore the focus to this element.
As of today focus will fall back to product link. However this link is removed soon. So it's nor clear were focus will end up.

**Comment by schuppell** [comment]
Discussed with Anthony:
--> The focusable element which comes before in the DOM should be selected after the button disappears.

**Comment by Uli-Tiger** [comment]
Ulrich Stellmacher  10:58 Uhr
wegen dem Focus handling:
Ich habe es jetzt etwas anders gelöst um die Komplexität nicht zu groß werden lassen.
99%-Fälle
Also es bekommt nun immer der vorherige Value den Focus. Soweit deckt sich ja das noch mit Antonys Wünschen.
Jetzt gibt es noch den Fall, daß ich den ersten Value selektiere. Nun bekommt der zweite Value den Fokus. Eigentlich sollte ja der letzte Value des vorherigen attributes den Fokus bekommen, aber das ist nicht so einfach, weil die Komponente gar nicht die ganze Konfiguration kennt => gößeres refactoring erforderlich. Außerdem kann es sein, daß es A) kein vorheriges Attribut gibt, B) das vorherige Attribute keine fokussierbaren Elemente enthält. Stattdessen den zweiten value zu nehmen ist da wesentlich simpler. Der Fokus bliebt also immer innerhalb des Attributes, was ja auch logiosch ist.
Sonderfälle:
Wenn der einzigste value selektiert wird, dann springt der focus nun wie bisher an den Anfang der Gruppe. Ich halte das aber eher für eine unsinnige Modellierung (mandatory selection with only one value!?! ) und würde da jetzt nicht investieren.
(bearbeitet)


Lukas Ralf Schuppel  11:11 Uhr
Find ich in Ordnug

**Comment by ChristophHi** [comment]
Has been fixed with https://github.com/SAP/spartacus/pull/11736
